### ISSUE TITLE
Handle entry-point failures gracefully

### ISSUE THREAD

**Issue by HarryMichal** [containers/toolbox#750]
**Is your feature request related to a problem? Please describe.**

Toolboxes can fail to start and in such situations users are not given any relevant information. The error happens in a process that is separate from execution of a parent command (e.g., `toolbox enter`). This fact is frustrating for users and annoying for developers who are trying to debug the cause of the fail.

**Describe the solution you'd like**

Extend the entry-point (`toolbox init-container`) so that it reports errors (both fatal and non-fatal) in a way that can be interpreted by the parent process. A place for this could be a lock file created in the runtime directory. The file creation currently means a container has been started correctly.

**Additional context**

Start synchronization is a known thing in the containers land and has to be handled somehow in e.g. CI systems. In Podman this is done by printing a specific string in the entry-point which is then inspected with `podman logs` or by attaching to the start of a container.

**Comment by debarshiray** [comment]
Wow!  It's been a while since @HarryMichal filed this, and even longer since we have been talking about it.

I have an initial implementation of this in https://github.com/containers/toolbox/pull/1496

> Extend the entry-point (toolbox init-container) so that it reports errors (both fatal and non-fatal) in a way that can be interpreted by the parent process.

It turns out that the `toolbox init-container` entry point already does what we need.  I had to hook up `podman logs` appropriately from `toolbox enter` and `run` to retrieve the debug logs and error strings, unmarshal them to understand the state of the entry point and re-display them to the user.

@HarryMichal already added some of this with `podman logs` to our system tests in September 2020 (commit [bbc23f3fb0358917](https://github.com/containers/toolbox/commit/bbc23f3fb03589176104c99e26b06ec092003729), pull request [#567](https://github.com/containers/toolbox/pull/567)).  So, this isn't something new, but it was time we folded it cleanly into the main `toolbox(1)` code.
